dvc/repo/__init__.py

```diff
@@ -375,6 +375,28 @@ def drop_data_index(self) -> None:
         self.data_index.commit()
         self._reset()
 
+    def get_data_index_entry(
+        self,
+        path: str,
+        workspace: str = "repo",
+    ) -> Tuple["DataIndex", "DataIndexEntry"]:
+        if self.subrepos:
+            fs_path = self.dvcfs.from_os_path(path)
+            fs = self.dvcfs.fs
+            # pylint: disable-next=protected-access
+            key = fs._get_key_from_relative(fs_path)
+            # pylint: disable-next=protected-access
+            subrepo, _, key = fs._get_subrepo_info(key)
```

Comment:
This subrepo lookup still feels like it belongs in `Repo` somewhere and not dvcfs, but at least this way the internal lookups are kept within the `Repo` level helper method and we don't need to touch the dvcfs internals in `dvc.api` methods